test/parallel/xyk-pallet.API.poolLiqidity.claimReward.test.ts

```diff
@@ -233,3 +251,52 @@ test("Given a pool with 2 users with activated rewards WHEN more than one period
   expect(differenceMGAUser1).bnGt(BN_ZERO);
   expect(differenceMGAUser2).bnGt(BN_ZERO);
 });
+
+test("Given a pool with user with activated rewards  WHEN it was deactivated AND activated again THEN the user can receive a reward in both a deactivated and activated pool", async () => {
+  await activateLiquidity(
+    testUser1.keyRingPair,
+    liqIdPromPool2,
+    Assets.DEFAULT_AMOUNT.divn(2)
+  );
+
+  await waitForRewards(testUser1, liqIdPromPool2);
+
+  await promotePool(sudo.keyRingPair, liqIdPromPool2, 0);
+
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+
+  await claimRewardsAll(testUser1, liqIdPromPool2).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+
+  await testUser1.refreshAmounts(AssetWallet.AFTER);
+
+  const deactivatedPoolRewards = testUser1
+    .getAsset(MGA_ASSET_ID)!
+    .amountAfter.free!.sub(
+      testUser1.getAsset(MGA_ASSET_ID)!.amountBefore.free!
+    );
+
+  await promotePool(sudo.keyRingPair, liqIdPromPool2, 20);
+
+  await waitForRewards(testUser1, liqIdPromPool2);
+
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+
+  await claimRewardsAll(testUser1, liqIdPromPool2).then((result) => {
```

Comment:
either use the regular claim reward or generate a user on every test. here you are claiming rewards for all the pools; perhaps previous tests already generated some.

Reply:
Merged to main branch. Fixed